scripts/contract-test.sh

```diff
@@ -0,0 +1,66 @@
+#!/bin/bash
+
+FKEY="faucet"
+TESTKEY="test"
+
+CHAINID=123
+MONIKER="localbenchmarktestnet"
+
+# remove existing daemon and client
+rm -rf ~/.emint*
+pkill -f "emint*"
+
+make install
+
+emintcli config keyring-backend test
+
+# Set up config for CLI
+emintcli config chain-id $CHAINID
+emintcli config output json
+emintcli config indent true
+emintcli config trust-node true
+
+# if $KEY exists it should be deleted
+emintcli keys add $FKEY
+
+# Set moniker and chain-id for Ethermint (Moniker can be anything, chain-id must be an integer)
+emintd init $MONIKER --chain-id $CHAINID
+
+# Use a custom genesis with pre-generated keys
+cp ./benchmark_ethmint_genesis.json $HOME/.emintd/config/genesis.json
+
+# Allocate genesis accounts (cosmos formatted addresses)
+emintd add-genesis-account $(emintcli keys show $FKEY -a) 1000000000000000000photon,1000000000000000000stake
+
+# Sign genesis transaction
+emintd gentx --name $FKEY --keyring-backend test
+
+# Collect genesis tx
+emintd collect-gentxs
+
+# Run this to ensure everything worked and that the genesis file is setup correctly
+emintd validate-genesis
+
+# Start the node (remove the --pruning=nothing flag if historical queries are not needed) in background and log to emintd.log
+emintd start --pruning=nothing --rpc.unsafe --log_level "main:info,state:info,mempool:info" --trace > emintd.log &
+
+sleep 1
+
+# Start the rest server with unlocked faucet key in background and log to emintcli.log 
+emintcli rest-server --laddr "tcp://localhost:8545" --unlock-key $FKEY --chain-id $CHAINID --trace > emintcli.log &
+
+solc --abi contracts/counter.sol --bin -o contracts/abi_bin
+abigen --bin=contracts/abi_bin/contracts_counter_sol_Counter.bin --abi=contracts/abi_bin/contracts_counter_sol_Counter.abi --pkg=counter --out=counter.go
```

Comment:
so I was originally thinking if we could use existing geth functionality to deploy a contract using plain Go, like this contract:

https://github.com/ethereum/go-ethereum/blob/9e88224eb86c92c78f5f03ffcd0793704a8b4309/contracts/checkpointoracle/contract/oracle.go#L35-L46

Reply:
will look into this, should be doable